js/modules/GLPI.js

```diff
@@ -0,0 +1,209 @@
+/**
+ * ---------------------------------------------------------------------
+ *
+ * GLPI - Gestionnaire Libre de Parc Informatique
+ *
+ * http://glpi-project.org
+ *
+ * @copyright 2015-2022 Teclib' and contributors.
+ * @copyright 2003-2014 by the INDEPNET Development Team.
+ * @licence   https://www.gnu.org/licenses/gpl-3.0.html
+ *
+ * ---------------------------------------------------------------------
+ *
+ * LICENSE
+ *
+ * This file is part of GLPI.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ * ---------------------------------------------------------------------
+ */
+
+import Clipboard from './Clipboard.js';
+import Dialogs from "./Dialogs.js";
+
+/**
+ * The core GLPI module code.
+ * This code should be loaded globally for all pages.
+ */
+
+// Save any previous value of GLPI.
+if (window.GLPI !== undefined) {
+    window._GLPI = window.GLPI;
+}
+// Create a window variable to loading modules from legacy scripts
+
+window.GLPILegacy = {
+    /** @type {{key: string, load_callback: function}[]} */
+    _load_queue: [],
+    loadModule(key, load_callback = undefined) {
+        if (load_callback === undefined) {
+            load_callback = () => {};
+        }
+        if (window.GLPI.registerKnownModule !== undefined) {
+            window.GLPI.registerKnownModule(key).then(load_callback);
+        } else {
+            this._load_queue.push({
+                key: key,
+                load_callback: load_callback
+            });
+        }
+    }
+};
```

Comment:
I don't recall if there was a specific need for this yet, but it was intended to be put here in case the module code is not loaded or ready yet.

You could queue a module for loafing and then when the main GLPI module gets loaded and ready, it would start processing everything that had been queued.

Reply:
`GLPILegacy` and `GLPI` are in same file, so it is nearly impossible to be able to call `GLPILegacy.loadModule()` when `GLPI.initialize()` has not been called yet.